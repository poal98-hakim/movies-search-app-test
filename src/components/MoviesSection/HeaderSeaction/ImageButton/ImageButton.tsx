import React from "react";
import "./ImageButton.scss"

interface IProps{
  icon: {src: string, alt: string};
  onClick: () => void;
  disabled: boolean;
}

const ImageButton : React.VFC<IProps> = ({ icon: {src, alt}, onClick, disabled }) => {
  return (
    <button className="image-button" onClick={onClick} disabled={disabled}>
        <img 
            src={src} 
            alt={alt} 
            width="25" 
        />
    </button>
  )
}
export default ImageButton;