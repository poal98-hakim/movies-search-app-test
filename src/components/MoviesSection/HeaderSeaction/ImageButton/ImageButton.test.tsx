import ImageButton from "./ImageButton";
import { render, screen } from '@testing-library/react';

const PROPS = {
    icon: {src: "../../images/right-arrow.png", alt: "left-arrow"},
    onClick: jest.fn(), 
    disabled: true
}

describe('ImageButton', () => {
  let props = {...PROPS};//using a shallow copy because I'm only reading the data. Otherwise due to the nested array I would do a deep copy

  beforeEach(() => { //before each test render the component and reset the value of props
    props = {...PROPS};
    render(<ImageButton {...props} />);
  });
  
  it('renders correctly', () => {
    const { baseElement } = render(<ImageButton {...props} />);
    expect(baseElement).toMatchSnapshot()
  })

  it('loads button', () => {
    expect(screen.getByRole("button")).toBeInTheDocument();
    expect(screen.getByRole("button")).toBeDisabled();
  })

  it('shows correctly icon', () => {
    expect(screen.getByAltText(props.icon.alt)).toBeInTheDocument();
    expect(screen.getByRole("img")).toHaveAttribute("src", props.icon.src);
  })

})
