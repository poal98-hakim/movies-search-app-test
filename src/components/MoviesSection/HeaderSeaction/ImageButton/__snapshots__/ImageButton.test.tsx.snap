// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageButton renders correctly 1`] = `
<body>
  <div>
    <button
      class="image-button"
      disabled=""
    >
      <img
        alt="left-arrow"
        src="../../images/right-arrow.png"
        width="25"
      />
    </button>
  </div>
  <div>
    <button
      class="image-button"
      disabled=""
    >
      <img
        alt="left-arrow"
        src="../../images/right-arrow.png"
        width="25"
      />
    </button>
  </div>
</body>
`;
